File: client/src/app/_models/jobParams.ts
import { User } from "./user";

export class JobsParams {
  title = '';
  jobType = '';
  selfPost = false;
  pageNumber = 1;
  pageSize = 5;
  genres?: string;
  skillsRequired?: string;
  city?: string;
  provinceOrState?: string;
  country?: string;
  orderBy = 'mostRecent';
  // --> Other sortig options:
  // -- "alphabetical" - ascending
  // -- "deadline" - desc.
  // -- "lastUpdated" - desc.

  // No need for the constructor
  constructor() {
  }
}

File: API/Helpers/JobParams.cs
namespace API.Helpers
{
    public class JobParams : PaginationParams
    {
        public string Title { get; set; }
        public string JobType { get; set; }
        public int? PosterID { get; set; }
        public bool SelfPost { get; set; }
        public string OrderBy { get; set; } = "mostRecent";
        public string Genres { get; set; }
        public string SkillsRequired { get; set; }
        public string City { get; set; }
        public string ProvinceOrState { get; set; }
        public string Country { get; set; }
    }
}